### PR TITLE
Bug 1830125: Add fallback create operand form if schema is missing or invalid

### PR DESCRIPTION
**Pull request by TheRealJon**
Add the 4.4 operand form component as a fallback so that CRD's without a structural schema don't render a blank form.
This required a few small updates to the old form component as well as some other small bug fixes that were discovered during implementation:
- Updated legacy create operand form component to accept same props as new component
- Update legacy form FieldGroup component so that it creates same DOM structure as new component. This is for test stability.
- Update legacy form html id attributes to match new form. Also test stability.
- Remove dead code `frontend/public/components/utils/expandable-section.tsx` and
`frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/field-group.tsx`. These were only used by the legacy form field group implementation.
- Fix edge case in operand util function `hasNoFields` where schemas with no valid type were returning false.
- Fix edge case in operand util function `hasNoFields` where `schema.additionalProperties` has a valid type, but function was returning true.
- Add unit test case for the above two fixes
- Fixed some magic strings

### CODE REVIEW

File: frontend/packages/console-shared/src/components/dynamic-form/utils.ts [hunk 1]
```diff
@@ -13,14 +13,6 @@ export const getSchemaErrors = (schema: JSONSchema6): SchemaError[] => {
           },
         ]
       : []),
-    ...(_.isArray(schema.type)
```
Comment (TheRealJon) [inline frontend/packages/console-shared/src/components/dynamic-form/utils.ts:16]: Swagger OpenAPI schema doesn't allow array types, so this check was not necessary.

**Comment by TheRealJon** [comment]
/hold

Something broke while rebasing

**Comment by TheRealJon** [comment]
/hold cancel

Fixed

**Comment by spadgett** [comment]
Do we have a long-term plan for removing the second operand form? It would be good to avoid maintaining two of these.

**Comment by TheRealJon** [comment]
@spadgett Yes, it will be removed in 4.6. In the 4.6 release, structural schemas will be required on CRD's, so this fallback will no longer be necessary.

**Comment by TheRealJon** [comment]
> @spadgett Yes, it will be removed in 4.6. In the 4.6 release, structural schemas will be required on CRD's, so this fallback will no longer be necessary.

@tlwu2013 correct me if I'm wrong.

File: frontend/packages/operator-lifecycle-manager/src/components/operand/const.ts [hunk 2]
```diff
@@ -11,18 +11,17 @@ export const DEFAULT_K8S_SCHEMA: JSONSchema6 = {
   type: SchemaType.object,
   properties: {
     metadata: {
-      type: 'object',
```
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/const.ts:14]: These are just some magic strings I think I missed on initial implementation.

File: frontend/packages/operator-lifecycle-manager/src/components/operand/utils.spec.ts [hunk 3]
```diff
@@ -14,22 +14,14 @@ import { HIDDEN_UI_SCHEMA } from './const';
 describe('getJSONSchemaOrder', () => {
```
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/utils.spec.ts:14]: Formatting and added another unit test to check an edge case in the `hasNoFields` helper.

File: frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts [hunk 4]
```diff
@@ -70,14 +62,14 @@ export const getDefaultUISchemaForPropertyName = (name) =>
 
 // Determine if a schema will produce an empty form field.
 export const hasNoFields = (jsonSchema: JSONSchema6): boolean => {
-  const type = getSchemaType(jsonSchema ?? {});
+  const type = getSchemaType(jsonSchema ?? {}) ?? '';
```
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts:65]: Default to empty string when type cannot be guessed and add new case in switch to handle it. If the schema type cannot be determined by `getSchemaType`, the schema will not render any fields.

File: frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts [hunk 5]
```diff
@@ -70,14 +62,14 @@ export const getDefaultUISchemaForPropertyName = (name) =>
...
   const handleArray = () => {
     return hasNoFields(jsonSchema.items as JSONSchema6);
   };
   const handleObject = () => {
     return (
       _.every(jsonSchema?.properties, hasNoFields) &&
-      _.every(jsonSchema?.additionalProperties as JSONSchema6, hasNoFields)
+      hasNoFields(jsonSchema?.additionalProperties as JSONSchema6)
```
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/utils.ts:72]: `additionalProperties` is either boolean or JSONSchema6. It doesn't have nested schemas like 'properties'.

**Comment by TheRealJon** [comment]
/hold

Updating html id's used in the old form component so that they match the new implementation. This will help reduce test flakes since the current integration tests us id selectors to get form input values.
